pkg/feature/segment/segment.go

```diff
@@ -34,10 +34,18 @@ type DownloadSegmentClient interface {
 	GetAll(ctx context.Context) ([]segments.Response, error)
```

Comment:
Just thinking about the folder structure, would it make more sense to call it `pkg/resource/segment/download.go` ?

I think that `feature` is not descriptive enough, and after we start moving in other (deploy, delete) stuff, the name segment also becomes misleading. WDYT?